snapcraft/tests/store_tests/test_download.py

```diff
@@ -0,0 +1,150 @@
+#
+# Copyright (C) 2016 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+
+from snapcraft import (
+    _store,
+    config,
+    storeapi,
+)
+from snapcraft.tests import store_tests
+
+
+class TestDownloadLogin(store_tests.TestCase):
+
+    def test_download_without_credentials(self):
+        self.assertRaises(storeapi.InvalidCredentialsError,
+                          self.download, 'basic')
+
+
+class TestSearchPackage(store_tests.TestCase):
+
+    def setUp(self):
+        super().setUp()
+        self.addCleanup(self.logout)
+        self.login()
```

Comment:
AIUI, anonymous searches is a workaround until macaroons are supported, so starting with the MP, no more anonymous searches. This specific search is for download anyway which was previously implemented with OAuth and as such was already requiring login.